### PR TITLE
DISPATCH-209 : linkroute test better for proactor

### PR DESCRIPTION
**Pull request by mgoulish**
I stole code from LinkRoute test in multi tenancy system test -- to make my linkroute test better for proactor -- don't hog control so long while checking for linkroute address to appear at the far router.

All 4 tests survived a 100-iteration test on my machine.

### CODE REVIEW

File: tests/system_tests_three_routers.py [hunk 1]
```diff
@@ -373,5 +433,189 @@ def run(self):
...
+            event.receiver.flow ( self.count )
+
+        if event.receiver == self.route_check_receiver:
+            self.proxy = RouterProxy ( self.route_check_receiver.remote_source.address )
+            self.poll()
+
+
+    def on_link_opening ( self, event ):
+        if event.receiver:
+            event.receiver.target.address = self.route_container
```
Comment (ted-ross) [inline tests/system_tests_three_routers.py]: self.route_container is a host address (0.0.0.0:<some-port>), not an AMQP address.  If you look at the trace of the test, you will see that the addresses in the paired attach frames are different.  This should not be the case.

File: tests/system_tests_three_routers.py [hunk 2]
```diff
@@ -373,5 +433,189 @@ def run(self):
...
+
+        if event.receiver == self.route_check_receiver:
+            response = self.proxy.response(event.message)
+            if response.status_code == 200 and (response.remoteCount + response.containerCount) > 0:
+                if self.route_check_timer :
+                    self.route_check_timer.cancel()
+                    self.route_check_timer = None
+                # After address appears at the far end of my router line, (router A)
+                # give more time to let it finish propagating.
+                time.sleep ( 2 )
```
Comment (ted-ross) [inline tests/system_tests_three_routers.py]: Is this sleep really necessary?  Why do you need to allow more time for propagation when you have just verified that the propagation is complete?

File: tests/system_tests_three_routers.py [hunk 3]
```diff
@@ -373,5 +433,189 @@ def run(self):
...
+
+    def on_timer_task(self, event):
+        self.parent.timeout()
+
+
+class LinkRoute ( MessagingHandler ):
+
+    """
+        Set up and use a link-route, to send a message this way:
+            receiver <--- A <--- B <--- C <--- sender
```
Comment (ted-ross) [inline tests/system_tests_three_routers.py]: More information is needed here.   Which endpoint is the initiator of the link?  Which endpoint is connected as the route_container?

**Comment by mgoulish** [comment]
new one coming shortly...

**Pull request closed without merging**